src/common/plugin/Artifacts.js

```diff
@@ -0,0 +1,47 @@
+/*globals define*/
+define([
+], function(
+) {
+    const Artifacts = function () {};
+
+    // Supposed to be used as an extension object, do not instantiate
+    Artifacts.prototype.constructor = null;
+
+    Artifacts.prototype.getArtifactsDir = async function() {
+        // Find the artifacts dir
+        const children = await this.core.loadChildren(this.rootNode);
+        return children.find(child => this.core.getAttribute(child, 'name') === 'MyArtifacts') ||
+            this.activeNode;
+    };
+
+    Artifacts.prototype.getAssetNameFromHash = async function(hash){
+        const metadata = await this.blobClient.getMetadata(hash);
+        if (metadata) {
+            return metadata.name.replace(/\.[^.]*?$/, '');
+        }
+    };
+
+    Artifacts.prototype.getAssetNameFromPath = async function (path) {
```

Comment:
Why is this fn in the shared helper when it isn't shared between both plugins?

Reply:
Moved this to child class.